examples/sugarscape.py

```diff
@@ -0,0 +1,261 @@
+# Sugarscape
+#
+# To run, you must install pygame:
+# pip install pygame
+#
+# Original paper:
+# Epstein, J.M. & Axtell, R. Artificial Life and Robotics (1997) 1: 33. https://doi.org/10.1007/BF02471109
+#
+# This reproduces the model according to NetLogo.
+#
+# The pygame visualization seems to have a busy wait loop which will peg a processor.
+#
+
+import argparse
+import dworp
+import logging
+import numpy as np
+import pygame
+from sugarscape_map import DEFAULT_SUGAR_MAP
+
+
+class Patch:
+    def __init__(self, max_sugar):
+        self.sugar = max_sugar
+        self.max_sugar = max_sugar
+
+    def growback(self):
+        # immediate growback model
+        # self.sugar = self.max_sugar
+
+        # constant growback model
+        self.sugar = min(self.sugar + 1, self.max_sugar)
+
+    def __repr__(self):
+        return 'Patch({}, {})'.format(self.sugar, self.max_sugar)
+
+
+class Turtle(dworp.SelfNamingAgent):
```

Comment:
Turtle is the name NetLogo gives to all agents. It's not specific to this paper. NetLogo is based on [Logo](https://en.wikipedia.org/wiki/Logo_(programming_language)) (the first computer language that I learned when I was 7). It had turtle graphics where you had this triangle on the screen and could issue commands like move or turn. As it moved it drew lines on the screen. It looked something like this: https://www.youtube.com/watch?v=mttyoaspa7U

Anyway, we should come up with a better name than turtle for the agents. Maybe SugarAgent